tests/snippets/builtin_range.py

```diff
@@ -108,6 +108,12 @@
 assert list(reversed(range(5, 0, -1))) == [1, 2, 3, 4, 5]
 assert list(reversed(range(1,10,5))) == [6, 1]
 
+# __reduce__
+assert range(10).__reduce__()[0] == range
+assert range(10).__reduce__()[1] == (0, 10, 1)
+assert range(10, 1, -2).__reduce__()[0] == range
+assert range(10, 1, -2).__reduce__()[1] == (10, 1, -2)
```

Comment:
You could try testing using `copy.copy()`; it should use `__reduce__` and return a `range` object that is `==` but not `is` the original